### PR TITLE
Tiger/penalty goalie tactic

### LINKED ISSUE #908: Make new PenaltyGoalie Tactic
### Description of the task

We made a very hacky PenaltyGoalie at RoboCup but it was very similar to the existing GoalieTactic and wasn't worth porting over. We should plan to make a proper PenaltyGoalie.

This ticket should be scoped and broken down if necessary. Specs for a PenaltyGoalie are needed (Penalty shot vs Penalty shootout?)

### Acceptance criteria

- [ ] TODO

### PR DESCRIPTION
**Pull request by xyztiger**
### Testing Done

still need to add tests

### Resolved Issues

resolves #908 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — COMMENTED** [review]
Can you modify the BUILD file so that I can build and see the simulated test in action?

**Comment by xyztiger** [comment]
> Can you modify the BUILD file so that I can build and see the simulated test in action?

Oh I actually didn't mean to include the simulated tests in this commit, could I get a review of my penalty_goalie_tactic.cpp and penalty_goalie_tactic_test.cpp first?

File: src/software/ai/hl/stp/tactic/penalty_goalie_tactic.cpp [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        double a1          = first.getCoeffs().a;
+        double b1          = first.getCoeffs().b;
+        double c1          = first.getCoeffs().c;
+        double a2          = second.getCoeffs().a;
+        double b2          = second.getCoeffs().b;
+        double c2          = second.getCoeffs().c;
+        double determinant = (a1 * b2) - (a2 * b1);
+
+        double x = ((b1 * c2) - (b2 * c1)) / determinant;
+        double y = ((a2 * c1) - (a1 * c2)) / determinant;
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/penalty_goalie_tactic.cpp]: seems like we should have a similar function in geom

File: src/software/ai/hl/stp/tactic/simulated_penalty_goalie_tactic_test.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class SimulatedPenaltyGoalieTacticTest : public SimulatedTacticTestFixture
+{
+};
+
+TEST_F(SimulatedPenaltyGoalieTacticTest, test_initial_position_on_goal_line) {
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/tactic/simulated_penalty_goalie_tactic_test.cpp:19]: You could set an initial velocity on the ball and check that is blocks it

**Review by jonathanlew — COMMENTED** [review]
I did an initial review on this PR but I'd really like to see simulated tests on it before diving deeper into it. How is this implementation different from the GoalieTactic, especially given there's only one attacker? Can you run fix formatting please? Please keep in mind that we want to move this to the Tactic FSM framework eventually

File: src/software/ai/hl/stp/tactic/penalty_goalie_tactic.cpp (hunk 1 above)
Comment (xyztiger) [reply src/software/ai/hl/stp/tactic/penalty_goalie_tactic.cpp]: I initially did this to get around having to use an optional pointer but I refactored it now to use intersection from geom

File: src/software/ai/hl/stp/tactic/simulated_penalty_goalie_tactic_test.cpp (hunk 2 above)
Comment (xyztiger) [reply src/software/ai/hl/stp/tactic/simulated_penalty_goalie_tactic_test.cpp:19]: Added this but I need some help setting with my tactic constructor

**Comment by akhilveeraghanta** [comment]
it looks like some random firmware changes made it into this PR, could you revert those and pull from upstream/master

that should fix the codespell/fixformatting problem when you run it again

**Pull request closed without merging**